pkg/project/project.go

```diff
@@ -358,6 +359,44 @@ func FromExactPath(path string) (*Project, error) {
 	return project, nil
 }
 
+// LocalProjectDoesNotExist is an error returned when a requested project is not checked out locally.
+type LocalProjectDoesNotExist struct{ error }
+
+// IsLocalProjectDoesNotExistError checks if the error is a LocalProjectDoesNotExist.
+func IsLocalProjectDoesNotExistError(err error) bool {
+	return errs.Matches(err, &LocalProjectDoesNotExist{})
+}
+
+// FromNamespaceLocal returns a local project (if any) that matches the given namespace.
+// This is primarily used by `state use` in order to fetch a project to switch to if it already
+// exists locally. The namespace may omit the owner.
+func FromNamespaceLocal(ns *Namespaced, cfg projectfile.ConfigGetter) (*Project, error) {
+	for namespace, paths := range projectfile.GetProjectMapping(cfg) {
+		if len(paths) == 0 {
+			continue
+		}
+		var namespaced Namespaced
+		err := namespaced.Set(namespace)
+		if err != nil {
+			logging.Debug("Cannot parse namespace: %v") // should not happen since this is stored
+			continue
+		}
+		if (!ns.AllowOmitOwner && strings.ToLower(namespaced.String()) == strings.ToLower(ns.String())) ||
+			(ns.AllowOmitOwner && strings.ToLower(namespaced.Project) == strings.ToLower(ns.Project)) {
+			return FromPath(paths[0]) // just pick the first one
```

Comment:
Heads up; filed this story to follow up:

https://activestatef.atlassian.net/browse/DX-1075

Felt out of scope for this PR.